ntbs-service/Pages/Index.cshtml.cs

```diff
@@ -81,10 +80,8 @@ private async Task SetUserAlertsAndTbServicesAsync()
             if (_userService.GetUserType(User) == UserType.NationalTeam)
                 return;
             var services = (await _userService.GetTbServicesAsync(User)).ToList();
-            var tbServiceCodes = services.Select(s => s.Code);
             TbServices = new SelectList(services, nameof(TBService.Code), nameof(TBService.Name));
-            var nonFilteredAlerts = await _alertRepository.GetOpenAlertsByTbServiceCodesAsync(tbServiceCodes);
-            Alerts = await _authorizationService.FilterAlertsForUserAsync(User, nonFilteredAlerts);
+            Alerts = await _authorizationService.FilterAlertsForUserAsync(User, _alertRepository.GetBaseOpenAlertIQueryable().ToList());
```

Comment:
I'm concerned by this - previously we were only getting the alerts for a user's TB services, but now we're fetching all the open alerts, and that could potentially be quite a lot.

Reply:
Yes I think this may be a bit of an issue - but I'm not sure how we can fix it. The problem is now we can't filter the alerts by TBService. If we were to filter alerts at this stage the logic would mirror the logic in `AlertIsForNotificationInOrTransferToGivenServices`, so I thought it best to only do it once.

Reply:
Could we change `FilterAlertsForUserAsync` so that it only does the user filtering (but not the TB Service filtering)?

Reply:
I've had a little look in the code, and this might be tricky since it's also used on the notification page... although we could conceivably move the logic to the alert repository? Doing it twice might be the simpler option, and we could put a comment in here that we're doing it for performance reasons.

Reply:
I'm trying this now. At the moment the `AlertIs...` function takes in an IQueryable<int> but we won't be able to do this in the alert repository as we will be using an alert queryable. This means either the list of alerts or list of notification IDs is going to have to be in memory doesn't it?

Reply:
I think it probably does. The list of notification IDs is probably not as bad, as each one is only a couple of bytes.

An alternative, would be to include the TB Service when we fetch the Alert, using a projection (i.e. instead of getting a list of `Alert`s we get a list of `AlertWithTbService` objects instead). That way we wouldn't need to fetch the notification IDs at all.